### ISSUE TITLE
[Date field] Unable to focus calendar once user has exited once

### ISSUE THREAD

**Issue by Magsu** [sl-design-system/components#2937]
### Provide a general summary of the issue here

Users are unable to move focus back to the calendar element after they've escaped it once. Users can close and open the calendar to get back to it, but this should not be intended behavior. Users should be able to move freely between the rest of the content and the calendar (or should be focus trapped to the calendar and not be able to move out of it).

The calendar element is in a `<slot>` tag. The tag has tabindex="-1" set to it, and while all the elements inside it do have tabindex="0", they are still not tabbable. This is most probably due to using shadowroot and how it currently delegates focus (i.e., children are not tabbable, because the tabindex prevents it). Setting tabindex="0" to the  `<slot>` allows the elements inside it to appear as normal in focus order (but the `<slot>` itself is still not tabbable). 

<img width="671" height="438" alt="Image" src="https://github.com/user-attachments/assets/01e43c3c-f0e7-4719-aa87-3f8c52f15179" />

### 🤔 Expected Behavior?

Users can move freely between the calendar and other elements using keyboard. 

### 😯 Current Behavior

When opened users focus is set to the calendar. When user leaves the calendar itself, they cannot focus back to the calendar.

### 💁 Possible Solution

Change tabindex="-1" to tabindex="0" in the `<slot>` tag should fix this (or could also prevent focus delegation in the shadow root).

### 🖥️ Steps to Reproduce

1. Open the calendar using keyboard
2. Tab out of the calendar using shift+tab
3. Press tab again -> focus does not move to the calendar

### What browsers are you seeing the problem on? (only for bugs in code)

Chrome

### What operating system are you using?

Windows 11

### 👤 Your name

Valtteri Järvinen

### 🧢 Your product/team

External

### 🎨 Your Theme(s)

Sanoma Learning
